includes/PluginFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * PluginFactory class.
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+use Google\Web_Stories\Infrastructure\ServiceBasedPlugin;
+
+/**
+ * PluginFactory class.
+ *
+ * The plugin factory is responsible for instantiating the plugin and returning
+ * that instance.
+ *
+ * It can decide whether to return a shared or a fresh instance as needed.
+ *
+ * To read more about why this is preferable to a Singleton,
+ *
+ * @see https://www.alainschlesser.com/singletons-shared-instances/
+ *
+ * @since 1.11.0
+ */
+class PluginFactory extends ServiceBasedPlugin {
```

Comment:
I have never understood the point of this class. A static method in a class seems a little pointless.

Reply:
One could look at it as the OOO version of the `get_plugin_instance()` function we initially had.

Reply:
The main benefit is autoloading. PHP is heavily optimized towards objects by now, and anything that is not an object is penalized both in functionality and performance.